resolver/resolver.cc

```diff
@@ -1786,16 +1786,34 @@ class ResolveSanityCheckWalk {
 };
 }; // namespace
 
-vector<ast::ParsedFile> Resolver::run(core::MutableContext ctx, vector<ast::ParsedFile> trees, WorkerPool &workers) {
+ast::MaybeASTPassResult Resolver::run(core::MutableContext ctx, vector<ast::ParsedFile> trees, WorkerPool &workers) {
```

Comment:
Is this granular enough? Only attempting to abort between macro resolver steps?

Also, would we want to push these checks into the respective functions, like resolveConstants or finalizeAncestors?

Reply:
Good question. I actually don't know because I don't know the runtime of those steps and if they are significant. From talking with Dmitry, resolver should be fairly fast, and most time should be in typechecking.